services/api-server/src/simcore_service_api_server/api/routes/solvers_jobs.py

```diff
@@ -56,41 +92,119 @@ async def create_job(
     catalog_client: CatalogApi = Depends(get_api_client(CatalogApi)),
+    webserver_api: AuthSession = Depends(get_webserver_session),
+    director2_api: DirectorV2Api = Depends(get_api_client(DirectorV2Api)),
     url_for: Callable = Depends(get_reverse_url_mapper),
 ):
     """Creates a job in a specific release with given inputs.
 
     NOTE: This operation does **not** start the job
     """
-    from .jobs_faker import create_job_impl
 
-    solver = await catalog_client.get_solver(user_id, solver_key, version)
-    return await create_job_impl(solver.id, solver.version, inputs, url_for)
+    async def _fake_impl():
+        from ._jobs_faker import create_job_impl
+
+        solver = await catalog_client.get_solver(user_id, solver_key, version)
+        return await create_job_impl(solver.id, solver.version, inputs, url_for)
+
+    async def _draft_impl():
+
+        # ensures user has access to solver
+        solver = await catalog_client.get_solver(user_id, solver_key, version)
+
+        # creates NEW job as prototype
+        pre_job = Job.create_solver_job(solver=solver, inputs=inputs)
+        logger.debug("Creating Job '%s'", pre_job.name)
+
+        # -> catalog
+        # TODO: validate inputs against solver input schema
+
+        #   -> webserver:  NewProjectIn = Job
+        project_in: NewProjectIn = create_new_project_for_job(solver, pre_job, inputs)
+        new_project: Project = await webserver_api.create_project(project_in)
+        assert new_project  # nosec
+        assert new_project.uuid == pre_job.id  # nosec
+
+        # for consistency, it rebuild job
+        job = create_job_from_project(
+            solver_key=solver.id,
+            solver_version=solver.version,
+            project=new_project,
+            url_for=url_for,
+        )
+        assert job.id == pre_job.id  # nosec
+        assert job.name == pre_job.name  # nosec
+
+        # -> director2:   ComputationTaskOut = JobStatus
+        # consistency check
+        task: ComputationTaskOut = await director2_api.create_computation(
+            job.id, user_id
+        )
+        assert task.id == job.id  # nosec
+
+        job_status: JobStatus = create_jobstatus_from_task(task)
+        assert job.id == job_status.job_id  # nosec
+
+        return job
+
+    return await _draft_impl()
 
 
 @router.get("/{solver_key:path}/releases/{version}/jobs/{job_id}", response_model=Job)
 async def get_job(
     solver_key: SolverKeyId,
     version: VersionStr,
     job_id: UUID,
+    webserver_api: AuthSession = Depends(get_webserver_session),
     url_for: Callable = Depends(get_reverse_url_mapper),
 ):
     """ Gets job of a given solver """
-    from .jobs_faker import get_job_impl
 
-    return await get_job_impl(solver_key, version, job_id, url_for)
+    async def _fake_impl():
+        from ._jobs_faker import get_job_impl
+
+        return await get_job_impl(solver_key, version, job_id, url_for)
```

Comment:
it's a bit confusing all these fake functions everywhere. but it will cleaned up right?

Reply:
in the next PR I will reuse them all for a fixture. For the moment I keep them in place. The idea is simple, every handle has TWO implementation functions ``_fake_impl`` and ``draft_impl`` inside the body

```python
async def my_handler():
    async def _fake_impl():
        ...
    async def _draft_impl():
       ...
    return _draft_impl()
```

For the moment, i tis convenient because i can change fast to the fake to compare examples/tests.